ModularTegustation/tegu_items/representative/console.dm

```diff
@@ -237,11 +237,112 @@
 			to_chat(usr, span_warning("ASSIGNMENT ERROR."))
 			playsound(get_turf(src), 'sound/machines/uplinkerror.ogg', 20, 1)
 			return
+	GoodieDelivery(our_corporation, user)
 	CreateResearchList(our_corporation)
 	name = "[our_corporation] corp representative console"
 	playsound(get_turf(src), 'sound/machines/terminal_success.ogg', 20, 1)
 	return
 
+/obj/structure/representative_console/proc/GoodieDelivery(our_corporation, mob/living/user)
+	//theres probably a super efficient method of doing this but Im following the method the console uses
+	var/corporationpick
+	if(our_corporation == K_CORP_REP)
+		corporationpick = /obj/item/storage/box/corporation/kcorp
+	if(our_corporation == L_CORP_REP)
+		corporationpick = /obj/item/storage/box/corporation/lcorp
+	if(our_corporation == R_CORP_REP)
+		corporationpick = /obj/item/storage/box/corporation/rcorp
+	if(our_corporation == W_CORP_REP)
+		corporationpick = /obj/item/storage/box/corporation/wcorp
+	if(our_corporation == N_CORP_REP)
+		corporationpick = /obj/item/storage/box/corporation/ncorp
+	if(our_corporation == P_CORP_REP)
+		corporationpick = /obj/item/storage/box/corporation/pcorp
+	if(our_corporation == J_CORP_REP)
+		corporationpick = /obj/item/storage/box/corporation/jcorp
```

Comment:
Don't, use a switch. That or maybe an associative list of defines to their objects
```suggestion
	switch(our_corporation)
		if(K_CORP_REP)
			corporationpick = /obj/item/storage/box/corporation/kcorp
		if(L_CORP_REP)
			corporationpick = /obj/item/storage/box/corporation/lcorp
		if(R_CORP_REP)
			corporationpick = /obj/item/storage/box/corporation/rcorp
		if(W_CORP_REP)
			corporationpick = /obj/item/storage/box/corporation/wcorp
		if(N_CORP_REP)
			corporationpick = /obj/item/storage/box/corporation/ncorp
		if(P_CORP_REP)
			corporationpick = /obj/item/storage/box/corporation/pcorp
		if(J_CORP_REP)
			corporationpick = /obj/item/storage/box/corporation/jcorp
```